### ISSUE TITLE
No underline color on active link with multiple <md-tab-group>

### ISSUE THREAD

**Issue by kevincaradant** [angular/material2#7274]
#### Bug, feature request, or proposal:
Use multiple `<md-tab>` and `<md-tab-group>` imbricated => lost the underline color on active `<md-tab>`

This is an example:

```
<div fxLayoutWrap fxLayoutGap="10px" fxLayout="row" fxLayout.xs="column" fxLayoutAlign="center start">
  <md-tab-group class="header-tab-admin" fxFlex="75">
    <md-tab label="{{'COMMONS.FAMILY.FAMILY' | translate}}"></md-tab>
    <md-tab label="{{'COMMONS.CREATE' | translate}}"></md-tab>
    <md-tab label="{{'COMMONS.USER.USER' | translate}}"></md-card>
    
    <md-tab label="{{'COMMONS.HOS.HOS' | translate}}"> // I select this tab
      <md-tab-group class="header-tab-2-sub-admin">
        <md-tab label="{{'COMMONS.SEARCH2' | translate}}"> // Here NO UNDERLINE COLOR UNTIL I CLICK ON IT
        </md-tab>
        <md-tab label="{{'COMMONS.CREATE2' | translate}}"></md-tab>
      </md-tab-group>
    </md-tab>
  </md-tab-group>
</div>
```

Picture:
![image](https://user-images.githubusercontent.com/6093175/30775685-58971b72-a099-11e7-99b8-48329dffb435.png)

#### What is the expected behavior?
An underline color on the active item <md-tab>

#### What is the current behavior?
No underline color for the active tab. See the snippet code mentioned before.

#### What is the use-case or motivation for changing an existing behavior?
I need to have two <md-tab> imbricated for my needs

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 4.3.6
OS: Windows 7 Pro 64 bits
TS: 2.4.1
Browser: Chrome V60

#### Is there anything else we should know?
Nothing else

**Comment by kevincaradant** [comment]
If I do a zoom- or zoom+ on the page or if I open the console of chrome, the underline appears again. 

Is it a kind of focus problem maybe ?

**Comment by julianobrasil** [comment]
Plunk to help describing it: https://plnkr.co/edit/qyKDATOdTb0msRnYjbbe?p=preview

**Comment by kevincaradant** [comment]
@julianobrasil , Thank you to have made the plunk. At the begin, I didn't read correctly, I believed that you asked to me to do it

**Comment by julianobrasil** [comment]
OK, copy the plunk link to the issue body description.

**Comment by crisbeto** [comment]
I'm having a hard time reproducing it from the provided Plunkr. Both the top and bottom levels of tabs are underlined.

**Comment by julianobrasil** [comment]
Maybe related to SO? I'm running it on a Windows 10 system, Google Chrome:

![image](https://user-images.githubusercontent.com/25405143/30822823-810d1238-a200-11e7-8f97-3a9742554114.png)

**Comment by crisbeto** [comment]
I'm running it on Windows with Chrome 60 as well.

**Comment by julianobrasil** [comment]
@crisbeto, my Chrome: Versão 61.0.3163.100 (Versão oficial) 64 bits

**Comment by crisbeto** [comment]
I just updated to Chrome 61 and I'm still not seeing it.

**Comment by julianobrasil** [comment]
I checked it out also in IE11 and Edge. I've changed the screen resolution to 3 different values (1024x768, 1280x1024 and 1920 x 1080) looking for any situation where the problem didn't occured. Unfortunately, I saw the problem in each case. 😞 

More info: this is not a new issue. This is a dupe of https://github.com/angular/material2/issues/5503.

Additionally, as mentioned above in https://github.com/angular/material2/issues/7274#issuecomment-331728260, if you change the browser zoom, the underline appears (the problem seems to be with the initial rendering):

![image](https://media.giphy.com/media/l378BkW2XBgcgiEF2/giphy.gif)

**Comment by julianobrasil** [comment]
I think the root case is the animations being broken in not selected tabs: #5269

**Comment by amcdnl** [comment]
Duplicate of #6478

**Comment by kevincaradant** [comment]
Any news about this issue ? :)

**Comment by amcdnl** [comment]
PR for fix is in queue, some other PRs need to get merged before it can go in though.

**Comment by karolanet333** [comment]
Hello!! I have the same issue. I'm experiencing this on a MacBook. Soy this issue is not relative to the SO as was suggested. I'd really appreciate if you could fix it. Thank you!!

**Comment by BeatriceThalo** [comment]
A workaround: whenever the outer mat-tab-group changes its selectedIndex to unhide your inner mat-tab-group, call window.dispatchEvent(new Event('resize'));

**Comment by Shreedevi-G-Rao** [comment]
Thanks it will work in chrome, but not in IE11.

**Comment by BeatriceThalo** [comment]
@Shreedevi-G-Rao True, IE needs a different method of creating the event. See this [SO answer](https://stackoverflow.com/a/49071358)

**Comment by nickkhan** [comment]
> A workaround: whenever the outer mat-tab-group changes its selectedIndex to unhide your inner mat-tab-group, call `window.dispatchEvent(new Event('resize'));`

I am using @Input from parent mat-tab-group to pass the selectedIndex to child component, seems to work well.

**Comment by irissoftworks** [comment]
I don't really get it how can this be a P3 ? As far as I understand there is no viable workaround ?

Our use case is that the tab groups are inside expandable rows within a mat table, therefore hidden by default. 
Dispatching the resize event may give users the opportunity to see the inkbar, but it also forces them to see the inkbar animation when the desired behaviour is that the inkbar is already in the correct position when the mat-table row is expanded. 

We are trying to deliver quality here, and so for our use case this generates a really bad visual experience because the tab content is immediately visible, while the inkbar takes a second or so to indicate the active tab. So now we have to compensate with an animation that we don't want in the tab content as well ? And this is the definition of a P3 ?

I have been using Angular Material for quite a while, used to love it and still kinda do, but unfortunately it seems that it has too many issues and too much complexity for the project capacity available. This causes that even nasty bugs like this one are all P3's and open for two years, and never get fixed. 

Would be great if someone could at least guide us if this is about to change or is it just going to be stacking up more and more of these issues ?

**Comment by francovp** [comment]
Any update on this?

**Comment by am-awais** [comment]
please fix that bug for God sake,
Mostly forums of angular material were ignored or just said its the copy of this copy of that, and no one gives the correct answer I don't know why the community is ignoring these major open issues.

**Comment by nicktobolski** [comment]
Also having this issue exactly. Wasted too many hours try to fix/work around it.

**Comment by stephen1807** [comment]
@noahch Thanks a lot! You saved me a lot of hours.

As others have said, please fix this!

**Comment by easycheese** [comment]
FYI lazy loading of tabs gets rid of this problem. Example nested tab group in tab 3

`<mat-tab-group mat-align-tabs="center" selectedIndex="0">
  <mat-tab label="Tab1">
    <p>
Tab 1 details
    </p>
  </mat-tab>
  <mat-tab label="Tab2">
    <p>
Tab 2 details
    </p>
  </mat-tab>
  <mat-tab label="Tab3" >
    <ng-template matTabContent>
      <mat-tab-group mat-align-tabs="center" selectedIndex="1" >
        <mat-tab label="Tab3-1">
          Test3-1
        </mat-tab>
        <mat-tab label="Tab3-2">
          Test3-2
        </mat-tab>
      </mat-tab-group>
    </ng-template>
    
  </mat-tab>
</mat-tab-group>`

**Comment by jacobweber** [comment]
If anyone needs help reproducing this, I'm happy to do so. Would love to get it fixed.

**Comment by roma2341** [comment]
> 

ngAfterViewInit works fine for me.

**Comment by Veargan** [comment]
My workaround for index mismatch which leads to wrong ink bar etc:
`
tabIndex: number; // bound to [selectedIndex] of mat-tab-group
@ViewChild(MatTabGroup, { static: false }) private matTabGroup: MatTabGroup;

setGroupIndex(): void { // call this function whenever you change your tabIndex
    if (this.tabIndex !== this.matTabGroup.selectedIndex) {
        this.matTabGroup.selectedIndex = this.tabIndex; 
    }
}
`

**Comment by zarend** [comment]
Hello @Veargan ,

Thank you for sharing a work-around. In order for the Angular team to take action on this issue, we would likely need a reproduction.

What version of Angular components does this issue happen for you on? And are you able to create a Stackblitz reproduction using a recent (ideally latest) version of Angular?

Best regards,

Zach
